### PR TITLE
add files_api for android java file

### PR DESCRIPTION
**Pull request by magaofei**
### What was a problem?

```
add  android java file API
add find content with java file API 
```

### How this PR fixes the problem?

```
DESCRIBE HERE
```

### Check lists (check `x` in `[ ]` of list items)

- [x] Run MobSF unit tests (http://your-mobsf-@ip:8000/tests/ or python3 manage.py test)
- [x] Tested Working on Linux, Mac, and Windows
- [x] Coding style (indentation, etc)

### Additional Comments (if any)

```
DESCRIBE HERE
```

### CODE REVIEW

File: StaticAnalyzer/views/android/find.py [hunk 1]
```diff
@@ -68,3 +68,55 @@ def run(request):
     except:
         PrintException("[ERROR] Searching Failed")
         return HttpResponseRedirect('/error/')
+
+
+def find_api(request):
```
[the remark this answers is not in the document]
Comment (magaofei) [reply StaticAnalyzer/views/android/find.py]: That may generate a lot of control statements

File: StaticAnalyzer/views/android/java.py [hunk 2]
```diff
@@ -60,3 +61,44 @@ def run(request):
     except:
         PrintException("[ERROR] Getting Java Files")
         return HttpResponseRedirect('/error/')
+
+
+def java_file_api(request):
```
[the remark this answers is not in the document]
Comment (magaofei) [reply StaticAnalyzer/views/android/java.py]: The original code is suitable for HTML, I can not modify it, if the code is modified, the maintainability of the code will be reduced, I try to get rid of some common code

[the remark this answers is not in the document]
Comment (ajinabraham) [reply StaticAnalyzer/views/android/java.py]: If you see the existing code base, we pass a default argument `isAPI=False` to every functions that serves both API and HTML. It won't be much work to have it the same way.

File: StaticAnalyzer/forms.py [hunk 3]
```diff
@@ -87,6 +87,33 @@ def clean_ext(self):
...
+class FindForm(forms.Form):
+    hash = forms.CharField(
+        max_length=32,
+        min_length=32
+    )
+    q = forms.CharField(min_length=1)
+    code = forms.ChoiceField(
+        choices=(
+            ('java', 'java'),
+            ('small', 'small')
```
Comment (ajinabraham) [inline StaticAnalyzer/forms.py]: Isn't it `smali` ?

File: StaticAnalyzer/views/android/find.py [hunk 4]
```diff
@@ -7,64 +7,75 @@
...
+            src = os.path.join(settings.UPLD_DIR, md5+'/smali_source/')
+            ext = '.smali'
+        else:
+            if is_api:
+                return HttpResponseBadRequest()
+            return HttpResponseRedirect('/error/')
+        # pylint: disable=unused-variable
+        # Needed by os.walk
+        for dir_name, sub_dir, files in os.walk(src):
+            if is_api and include_path and len(include_path) > 0:
```
Comment (ajinabraham) [inline StaticAnalyzer/views/android/find.py]: What is this code doing?

Comment (magaofei) [reply StaticAnalyzer/views/android/find.py]: Usually the entire directory will be scanned. This parameter can specify the directory.

File: StaticAnalyzer/forms.py (hunk 3 above)
Comment (magaofei) [reply StaticAnalyzer/forms.py]: fixed it

File: StaticAnalyzer/views/android/find.py (hunk 4 above)
Comment (magaofei) [reply StaticAnalyzer/views/android/find.py]: This is an optional parameter. The purpose of this parameter is to reduce the return of unnecessary information, such as only looking for the content under the current app, removing some third-party content.

**Comment by ajinabraham** [comment]
API Docs are not updated
The naming of API is confusing. The APIs should have meaningful names

**Comment by magaofei** [comment]
Do you have any advice? About the name of the API @ajinabraham

**Comment by ajinabraham** [comment]
@magaofei I will get back. Travelling this whole week.

**Comment by ajinabraham** [comment]
The API name can be something like
`/list_files`
and take params as androd, ipa, zip etc...

**Comment by ajinabraham** [comment]
Closing as the PR is not active.

**Pull request closed without merging**